### PR TITLE
[Doc Rework] Other changes PR5

### PR DESCRIPTION
**Pull request by mapedraza**
This PR is part of a bigger series https://github.com/telefonicaid/fiware-orion/pull/4141

- [ ] ~3. Metadata update semantics~ Done in previous PR
- [ ]     -~What about the rest of metadata.md document? Check that it is somehow covered in the current orion-api.md document. I think is is mostly covered, in negative case include it (maybe in "Context Metadata" subsection)~
- [x] 13. Subscription payload validations
   - [x]     - The information in this section has been already captured by the "Optional" and "Type" columns in the subscription metadata
   - [x]     - In addition, it is pretty much obsolete
   - [x]     - Thus, it can be just removed
- [x] 38. Deprecated features
   - [x]      - Unsure. Maybe to be fully removed. Leave it for the last PR ;)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -183,9 +183,6 @@ Reference sections below in order to know more details about the API
     [(en)](http://telefonicaid.github.io/fiware-orion/api/v2/stable/cookbook)
     [(jp)](https://open-apis.letsfiware.jp/fiware-orion/api/v2/stable/cookbook)
     (Apiary)
-    -   See also NGSIv2 implementation notes
```
Comment (fgalan) [inline README.md:186]: Now that README.md has been touched, maybe we can take the opportunity to include the link to the new document (just after the bullet about the walkthrough)

Same in the "Reference Documentation" section

(Not sure what to do with the links to the cookbook. Probably they are obsolete, but leave them by the moment.)

File: doc/manuals/admin/diagnosis.md [hunk 2]
```diff
@@ -258,7 +258,7 @@ Some possible values for `lastFailureReason` (non exahustive list):
 
 More detail about `status`, `lastFailureReason` and `lastSuccessCode` in the
```
Comment (fgalan) [inline doc/manuals/admin/diagnosis.md:259]: Replace the reference to NGSIv2 spec by the reference to Orion API

No direct reference to NGSIv2 spec should remain in the documentation. The reference in the Orion API preface should suffice. I'll add this as an item to #4141

File: doc/manuals/admin/cli.md [hunk 3]
```diff
@@ -118,7 +118,7 @@ The list of available options is the following:
     for forwarding messages and for notifications. Default timeout (if this parameter is not specified)
     is 5000 (5 seconds). Max value is 1800000 (30 minutes). This parameter can be defined individually for subscriptions. If defined on the subscription's
     JSON, the default parameter would be ignored. See section in the 
-    [NGSIv2 Implementation Notes](../user/ngsiv2_implementation_notes.md#timeout-subscriptions-option)
+    [`subscription.notification.htt`](../orion-api.md#subscriptionnotificationhttp).
```
Comment (fgalan) [inline doc/manuals/admin/cli.md:121]:
```suggestion
    [`subscription.notification.http`](../orion-api.md#subscriptionnotificationhttp).
```

File: doc/manuals/orion-api.md [hunk 4]
```diff
@@ -3198,17 +3198,17 @@ A `subject` contains the following subfields:
...
-| [`condition`](#subscriptionsubjectcondition) | ✓        | object | Condition to trigger notifications. If omitted, it means "any attribute change will trigger condition" |
+| `entities`                                   | ✓        | string | A list of objects, each one composed of the following subfields: <ul><li><code>id</code> or <code>idPattern</code> id or pattern of the affected entities. Both cannot be used at the same time, but one of them must be present.</li> <li><code>type</code> or <code>typePattern</code> Type or type pattern of the affected entities. Both cannot be used at the same time. If omitted, it means "any entity type".</li></ul> These fields should fllow the restrictions for IDs and types, be a valid regex (if pattern), and be not empty. |
```
Comment (fgalan) [inline doc/manuals/orion-api.md:3201]:
```suggestion
| `entities`                                   | ✓        | string | A list of objects, each one composed of the following subfields: <ul><li><code>id</code> or <code>idPattern</code> id or pattern of the affected entities. Both cannot be used at the same time, but one of them must be present.</li> <li><code>type</code> or <code>typePattern</code> Type or type pattern of the affected entities. Both cannot be used at the same time. If omitted, it means "any entity type".</li></ul> These fields should follow the restrictions for IDs and types, be a valid regex (if pattern), and be not empty. |
```

In additon, add a link to the section about identifiers restrictions.

File: doc/manuals/orion-api.md [hunk 5]
```diff
@@ -3267,7 +3267,7 @@ A `http` object contains the following subfields:
 
 | Parameter | Optional | Type   | Description                                                                                   |
 |-----------|----------|--------|-----------------------------------------------------------------------------------------------|
-| `url`     |          | string | URL referencing the service to be invoked when a notification is generated. An NGSIv2 compliant server must support the `http` URL schema. Other schemas could also be supported. |
+| `url`     |          | string | URL referencing the service to be invoked when a notification is generated. It mus follow the `http` URL schema. |
```
Comment (fgalan) [inline doc/manuals/orion-api.md:3270]:
```suggestion
| `url`     |          | string | URL referencing the service to be invoked when a notification is generated. It must follow the `http` URL schema format. |
```

Comment (fgalan) [reply doc/manuals/orion-api.md:3270]: Algo de `https` schema format would be valid?

File: doc/manuals/orion-api.md [hunk 6]
```diff
@@ -4213,3 +4213,24 @@ for the following aspects:
 * The `expired` value for `status` is not supported. The status is shows as `active` even in the
    registration is actually expired.
 * `legacyForwarding` field (within `provider`) to support forwarding in NGSIv1-based query/update format for legacy Context Providers 
+
+## Deprecated features
+
+Although we try to minimize the changes in the stable version of the NGSIv2 specification, a few changes
+have been implemented since the first release. Thus, there is changed functionality that doesn't appear in the current
+NGSIv2 stable specification document but that Orion still supports
+(as [deprecated functionality](../deprecated.md)) in order to keep backward compatibility.
```
Comment (fgalan) [inline doc/manuals/orion-api.md:4222]:
```suggestion
During the NGSIv2 specification work, some features were developed but they don't remain in the final version of the specification. However, Orion implemented them at some version and are still supported in order to keep backward compatibility (as [deprecated functionality](../deprecated.md)). They are documented here, although **you are highly encouraged to not use any of them**.
```

File: doc/manuals/orion-api.md (hunk 5 above)
Comment (mapedraza) [reply doc/manuals/orion-api.md:3270]: I think It is not needed to mention https

File: doc/manuals/admin/diagnosis.md (hunk 2 above)
Comment (mapedraza) [reply doc/manuals/admin/diagnosis.md:259]: Done in caaddeea98a0253ca07fad4cc04036d04b1fc3a0

File: doc/manuals/orion-api.md (hunk 4 above)
Comment (mapedraza) [reply doc/manuals/orion-api.md]: Done in 6e68e0f9f2e7b03701a6e5864b1026f35fa93991

File: README.md (hunk 1 above)
Comment (mapedraza) [reply README.md:186]: I think this is better for the last task in order to replace the references to specification or additionally include the reference for the specification.

Comment (fgalan) [reply README.md:186]: Ok. NTC

**Pull request merged**